core/meta/inc/TClass.h

```diff
@@ -374,7 +374,11 @@ friend class ROOT::Internal::TCheckHashRecursiveRemoveConsistency;
    TMethod           *GetClassMethodWithPrototype(const char *name, const char *proto,
                                                   Bool_t objectIsConst = kFALSE,
                                                   ROOT::EFunctionMatchMode mode = ROOT::kConversionMatch);
-   Version_t          GetClassVersion() const { fVersionUsed = kTRUE; return fClassVersion; }
+   Version_t          GetClassVersion() const {
+      if (!fVersionUsed.load(std::memory_order_relaxed))
```

Comment:
My understanding of `std::memory_order_relaxed` is it will give you a value that was the value set by  some thread at some time. So for this case that is fine since setting `VersionUsed` again would not be a problem.